### PR TITLE
Fixed #10107 -- Allowed mark_safe() to be used as decorator

### PR DESCRIPTION
**Pull request by ArcTanSusan**
I cherry-picked old patch written by matehat 7 yrs ago. All unit tests pass.

Original ticket: https://code.djangoproject.com/ticket/10107

This is part of the djangocon europe 2016 sprints. :saxophone: As always, feedback is welcome and I usually respond within 24 hours.

### CODE REVIEW

File: django/utils/safestring.py [hunk 1]
```diff
@@ -114,13 +114,24 @@ def _proxy_method(self, *args, **kwargs):
...
 def mark_safe(s):
     """
     Explicitly mark a string as safe for (HTML) output purposes. The returned
     object can be used everywhere a string or unicode object is appropriate.
 
+    If used on a method as a decorator, mark the returned data as safe.
+
     Can be called multiple times on a single string.
     """
+    if callable(s):
```
Comment (funkybob) [inline django/utils/safestring.py]: If a callable has a __html__ method, won't this break things?

File: tests/decorators/tests.py [hunk 2]
```diff
@@ -90,6 +97,39 @@ def test_attributes(self):
...
+        """
+        Tests that safety markers from django.utils.safestring
+        return the proper str or unicode subclass for use in templates.
+
+        Tests are done by comparing directly a sample rendered Template instance
+        with the unicode content it should contain.
+        """
+        template = Template("{{ data }}")
+
+        rendered = {'safe': u'<html><body>dummy</body></html>',
```
Comment (timgraham) [inline tests/decorators/tests.py]: no u'' prefixes please (use `from __future__ import unicode_literals` if needed)

File: docs/ref/utils.txt [hunk 3]
```diff
@@ -826,6 +826,8 @@ appropriate entities.
 
     Can be called multiple times on a single string.
 
+    Can also be used as a decorator.
```
Comment (timgraham) [inline docs/ref/utils.txt]: Need a `.. versionchanged:: 1.10` annotation and a mention in the release notes.

**Comment by timgraham** [comment]
buildbot, add to whitelist.

**Comment by svvitale** [comment]
This PR can be closed, replaced with https://github.com/django/django/pull/6706

**Pull request closed without merging**